Item and Weapon store

Kavignon opened the pull request:
A basic weapon and store system for when players will want to either acquire items or sell items in order to buy more items or better items. In a next development cycle, a character store would be added and would have similar functionalities with the weapon/item store.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gameoflightandshadows/a-game-of-light-and-shadows/103)
<!-- Reviewable:end -->

Commit:
#19 - Added an item category to make it easy to find out the type of item that was bought from the store. From there, when the option of sending the item to the active character is done by putItemInEquipment

review-ninja commented:
<a href="https://app.review.ninja/GameOfLightAndShadows/A-game-of-light-and-shadows/pull/103" target="_blank"><img src="https://app.review.ninja/66727011/pull/103/badge" alt="ReviewNinja"/>

Commit:
#19 - Create domain model of the store’s manager state and a player’s transaction. Every message that will be received by the manager are there too.

Commit:
#19 - Added the functionality for a transaction in the shop. Type system breaking because of file order which will be fixed over the next commits.
Items can now be removed/added to the transaction while updating the current bill. Later on, once the confirmation is received, then the inventory/character will see the update with the new item(s).

Kavignon commented:
Merging now so the fix the build with FixBuild branch. The branch will remain alive until the work is over.
